### PR TITLE
Added Tutorial Dialogue + Highlight

### PR DESCRIPTION
**Pull request by he-larious**
See issue #28 for specifics. Also this is coming from the Functions branch and not Tutorial due to merge conflicts.

### CODE REVIEW

File: Scripts/Level_Components/IDE.gd
Comment (m-lin1) [inline Scripts/Level_Components/IDE.gd:1]: Need to change this back to the one in Main or Combine

Comment (he-larious) [reply Scripts/Level_Components/IDE.gd:1]: Jokes on you, the entire thing was just an old duplicated file :)

File: Scripts/Singletons/DialogueManager.gd [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	var path_to_highlight = highlight_path[dialogue]
+	var component_highlight = level.get_node(path_to_highlight)
+	
+	component_highlight.visible = true
+	return component_highlight
+
+# Checks if user completed the correct action before the next line of dialogue is triggered
+func dialogue_progress_check(level):
+	for i in range(level.progress_check.size()):
+		if !level.progress_check[i] and fba_children_check(level.MainFBA, level.progress_check_arr[i]):
```
Comment (m-lin1) [inline Scripts/Singletons/DialogueManager.gd:66]: Might want to make this more general so the FBA parameter can be any FBA (i.e. for Recursion levels we might want to drag an F1 code block into the F1 IDE section

**Review by m-lin1 — COMMENTED** [review]
Tutorial Level 2 has a bug where the Dialogue starts again after pressing space on the finished level dialogue.

Doing a user action too early stops the next dialogue from popping up (i.e. Dialogue says to put Forward code block into Main, but I do that before pressing space to make the Dialogue move on. Now even if I remove Forward and put it back into Main, it won't trigger the next dialogue option) -> FIXED

**Review by m-lin1 — APPROVED** [review]

**Pull request merged**